MonoDevelop.MSBuild.Editor/Completion/MSBuildCompletionSource.cs

```diff
@@ -107,12 +123,132 @@ async Task<MSBuildCompletionSessionContext> GetSessionContext (IAsyncCompletionS
 			return new CompletionContext (ImmutableArray<CompletionItem>.Empty.AddRange (items));
 		}
 
+		//Dictionary<int, string> cache = new Dictionary<int, string> ();
+
+		protected override async Task<CompletionContext> GetAttributeValueCompletionsAsync (
```

Comment:
Overriding `GetAttributeValueCompletionsAsync` is redundant as the `MSBuildValueKind.NuGetID` and `MSBuildValueKind.NuGetVersion` cases in `GetExpressionCompletionsAsync` already handle triggering package name and version completion in attribute values (when the schema is loaded).

Reply:
The implementation of this method is good work, however :)